### ISSUE TITLE
-ddump-hi not passed to GHCJS

### ISSUE THREAD

**Issue by snoyberg** [commercialhaskell/stack#2718]
### Steps to reproduce

1. Start a GHCJS project
2. `stack build`
3. `find .stack-work -name \*dump\*`

### Expected

Some HI dump files would be found.

### Actual

No dump files are created, since Cabal does not pass -ddump-hi to GHCJS. The reason is that, in Stack.Build.Execute, we use `--ghc-options`, whereas Cabal expects us to use `--ghcjs-options`. Note that @nh2 points out that this is inconsistent in Cabal, since the .cabal file still respects `ghc-options:` whereas the command line flag does not work.

Workaround: add `ghc-options: -ddump-hi -ddump-to-file` to your .cabal file.

### Stack version

```
$ stack --version
Version 1.2.0, Git revision 241cd07d576d9c0c0e712e83d947e3dd64541c42 (4054 commits) x86_64 hpack-0.14.0
```

**Comment by nh2** [comment]
I've asked the question on `--ghc-options` vs `--ghcjs-options` at https://github.com/haskell/cabal/issues/3999

**Comment by snoyberg** [comment]
Looks like @mgsloan discovered and fixed this already: #2714
